models.py

```diff
@@ -0,0 +1,350 @@
+        rides = db_cursor.fetchall()
+        for ride in rides:
+            if ride[0] == ride_id:
+                db_cursor.execute("UPDATE rides SET ride=%s, driver_id=%s, departuretime=%s, numberplate=%s, maximum=%s WHERE ride_id=%s",
+                                  (ride, driver_id, departuretime, numberplate, maximum, ride_id))
+                db_connection.commit()
+                db_connection.close()
+
+                return make_response(jsonify({"message" : "ride has been successfully updated"}), 200)
+        return make_response(jsonify({"message" : "ride does not exist"}), 404)
+
+    @staticmethod
+    def start_ride(ride_id, driver_id):
+        """starts a ride"""
+        db_connection = conn()
+        db_cursor = db_connection.cursor()
+        db_cursor.execute("SELECT * FROM rides WHERE ride_id=%s", (ride_id,))
+        ride = db_cursor.fetchall()
+        if ride != []:
+            ride = ride[0]
+            if int(ride[2]) == driver_id:
+                db_cursor.execute("UPDATE rides SET status=%s WHERE ride_id=%s", ("given", ride_id))
+                db_connection.commit()
+                db_connection.close()
+
+                return {"message" : "ride has started"}
+
+            return {"message" : "The ride you want to start is not your ride."}
+
+        return {"message" : "ride does not exist"}
+
+    @staticmethod
+    def delete_ride(ride_id):
+        """Deletes a ride"""
+        db_connection = conn()
+        db_cursor = db_connection.cursor()
+        db_cursor.execute("SELECT * FROM rides")
+        rides = db_cursor.fetchall()
+
+        for ride in rides:
+            if ride[0] == ride_id:
+                db_cursor.execute("DELETE FROM rides WHERE ride_id=%s", (ride_id,))
+                db_connection.commit()
+                db_connection.close()
+
+                return make_response(jsonify({"message" : "ride has been successfully deleted"}), 200)
+        return make_response(jsonify({"message" : "user does not exists"}), 404)
+
+    @staticmethod
+    def get_ride(ride_id):
+        """Gets a particular ride"""
+        db_connection = conn()
+        db_cursor = db_connection.cursor()
+        db_cursor.execute("SELECT * FROM rides WHERE ride_id=%s", (ride_id,))
+        ride = db_cursor.fetchall()
+        db_connection.close()
+
+        if ride != []:
+            ride=ride[0]
+            info = {ride[0] : {"ride": ride[1],
+                                "driver_id": ride[2],
+                                "departure_time": ride[3],
+                                "cost": ride[4],
+                                "maximum": ride[5],
+                                "status": ride[6]}}
+            return make_response(jsonify({"ride" : info}), 200)
+        return make_response(jsonify({"message" : "ride does not exists"}), 404)
+        
+    @staticmethod
+    def get_all_rides():
+        """Gets all rides"""
+        db_connection = conn()
```

Comment:
refactor this to prevent opening a connection for every single request. You can move this to where your app is being initialized.